### PR TITLE
sched/cpuload: Support snapshot in specific time interval

### PR DESCRIPTION
**Pull request by jeongarmy**
Support CPU load measurement in specific time interval.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [90ce912393a0f076d50bfa10e37fed0bde268727] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [90ce912393a0f076d50bfa10e37fed0bde268727] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a6b37a120b27a9fb716128896b4555c354bf80d8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a6b37a120b27a9fb716128896b4555c354bf80d8] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3afae75508196d95fd39d22bd334d01b9ec818f9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3afae75508196d95fd39d22bd334d01b9ec818f9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [2bd41d85b6cb29a85c5d96a21e4101c5c123d661] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2bd41d85b6cb29a85c5d96a21e4101c5c123d661] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b0388780200de0b3f80f31e8c49442cf9d194e48] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b0388780200de0b3f80f31e8c49442cf9d194e48] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a65f223d051d06bedf63635244897c3183d7b7dc] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a65f223d051d06bedf63635244897c3183d7b7dc] - Code Rule Check (C++) OK.

File: os/kernel/sched/sched_cpuload.c [hunk 1]
```diff
@@ -273,4 +306,49 @@ int clock_cpuload(int pid, int index, FAR struct cpuload_s *cpuload)
...
+	hash_index = PIDHASH(pid);
+	DEBUGASSERT(cpuload);
+
+	/* Check if the entry is valid (TCB field is not NULL) */
+	if (g_pidhash[hash_index].tcb == NULL) {
+		return -ESRCH;
+	}
+
+	cpuload->active = 0;
+	cpuload->total = g_cpusnap_ticks;
```
Comment (sunghan-chang) [inline os/kernel/sched/sched_cpuload.c]: When arr is not full-filled, is it ok to use the size of arr?

File: os/kernel/sched/sched_cpuload.c [hunk 2]
```diff
@@ -110,13 +110,40 @@ static volatile uint32_t g_cpuload_timeconstant[SCHED_NCPULOAD] = {
 #endif
 };
 
+static pid_t *g_cpusnap_arr;
+static int16_t g_cpusnap_head;
+static int16_t g_cpusnap_ticks;
```
Comment (sunghan-chang) [inline os/kernel/sched/sched_cpuload.c]: arr_size

File: os/kernel/sched/sched_cpuload.c (hunk 1 above)
Comment (jeongarmy) [reply os/kernel/sched/sched_cpuload.c]: yes, cpuload monitor is started after snapshot interval (g_cpusnap_ticks) when snapshot mode.

**Comment by seinfra** [comment]
Target : [4a23f05c96398e21c33d7d53e29b8e53ecb0e5a8] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4a23f05c96398e21c33d7d53e29b8e53ecb0e5a8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8473d98f1b6305d7d5ec4ac75953b1ff11475559] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [8473d98f1b6305d7d5ec4ac75953b1ff11475559] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [51ff46ef4bcb2f65e2b4e0d6df3390ddd3acc0ae] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [51ff46ef4bcb2f65e2b4e0d6df3390ddd3acc0ae] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [710127b144d7c60c73d6e31c28795dded14583ff] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [710127b144d7c60c73d6e31c28795dded14583ff] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c29480a8c7763f7b71dfcf7345c69c43f261bbd3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c29480a8c7763f7b71dfcf7345c69c43f261bbd3] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b9bd047c9a786165cbfd3ebfee2b14c99e14e34f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b9bd047c9a786165cbfd3ebfee2b14c99e14e34f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0e5683b6b50616bdcb445e547b90885c505ad56a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0e5683b6b50616bdcb445e547b90885c505ad56a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cbc48efe42058254a075dad9f23745ab55130b68] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [cbc48efe42058254a075dad9f23745ab55130b68] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9bbaa517e2addfd5b95b0fac9d58f1677c526f59] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9bbaa517e2addfd5b95b0fac9d58f1677c526f59] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [5a9c426eeb72fbcb080e8f374c79746fba27d58a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5a9c426eeb72fbcb080e8f374c79746fba27d58a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [084b470f2ef55e7af729598b91b8cf7765c49df1] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [084b470f2ef55e7af729598b91b8cf7765c49df1] - Code Rule Check OK.

**Pull request merged**